tests/test_job.py

```diff
@@ -436,3 +436,274 @@ def update_status(self) -> JobStatus:
+        Job._build_dependency_string(job)
+
+
+def test_build_dependency_string_with_unsubmitted_job_in_list_raises() -> None:
+    """Test that using an unsubmitted job in a list raises ValueError."""
+    job1 = Job(Path("script1.py"), name="franck")
+    job1.job_id = "12345"
+
+    job2 = Job(Path("script2.py"), name="sinatra")
+    job2.job_id = None
+
+    with pytest.raises(ValueError, match="Job 'sinatra' has not been submitted yet"):
+        Job._build_dependency_string([job1, job2])
+
+
+def test_build_dependency_string_with_different_dependency_types() -> None:
+    """Test building dependency strings with different dependency types."""
+    job = Job(Path("script.py"))
+    job.job_id = "12345"
+
+    assert Job._build_dependency_string(job, "afterok") == "afterok:12345"
+    assert Job._build_dependency_string(job, "afterany") == "afterany:12345"
+    assert Job._build_dependency_string(job, "afternotok") == "afternotok:12345"
+    assert Job._build_dependency_string(job, "after") == "after:12345"
+
+
+def test_submit_pbs_with_string_dependency(
+    tmp_path: Path, monkeypatch: pytest.MonkeyPatch
+) -> None:
+    """Test submitting a job with a string job ID dependency."""
+    script = tmp_path / "script.py"
+    script.write_text("")
+    outdir = tmp_path
+    job = Job(script, name="dependent_job", output_folder=outdir)
+    pbs_path = tmp_path / "dependent_job.pbs"
+    job.write_pbs(pbs_path)
+
+    captured_cmd = []
+
+    class Dummy:
+        def __init__(self) -> None:
+            self.stdout = "99999.server\n"
+            self.stderr = ""
+
+    def mock_run(cmd, *args, **kwargs):
+        captured_cmd.extend(cmd)
+        return Dummy()
+
+    monkeypatch.setattr(subprocess, "run", mock_run)
+
+    job.submit_pbs(depend_on="12345.datarmor")
+
+    assert "-W" in captured_cmd
+    w_index = captured_cmd.index("-W")
+    assert captured_cmd[w_index + 1] == "depend=afterok:12345.datarmor"
+    assert job.job_id == "99999"
+
+
+def test_submit_pbs_with_job_dependency(
+    tmp_path: Path, monkeypatch: pytest.MonkeyPatch
+) -> None:
+    """Test submitting a job that depends on another Job instance."""
+    script = tmp_path / "script.py"
+    script.write_text("")
+    outdir = tmp_path
+
+    job1 = Job(script, name="job1", output_folder=outdir)
+    job1.job_id = "12345"
+
+    job2 = Job(script, name="job2", output_folder=outdir)
+    pbs_path = tmp_path / "job2.pbs"
+    job2.write_pbs(pbs_path)
+
+    captured_cmd = []
+
+    class Dummy:
+        def __init__(self) -> None:
+            self.stdout = "67890.server\n"
+            self.stderr = ""
+
+    def mock_run(cmd, *args, **kwargs):
+        captured_cmd.extend(cmd)
+        return Dummy()
+
+    monkeypatch.setattr(subprocess, "run", mock_run)
+
+    job2.submit_pbs(depend_on=job1)
+
+    assert "-W" in captured_cmd
+    w_index = captured_cmd.index("-W")
+    assert captured_cmd[w_index + 1] == "depend=afterok:12345"
+    assert job2.job_id == "67890"
```

Comment:
It feels like using a parametrized test would be more efficient here as you have quite long tests that do similar things with quite a bit of repeated code lines, right?